components/api/core/api_impl.bal

```diff
@@ -260,56 +260,78 @@ public function getImageByImageName(http:Request getImageRequest, string orgName
 # + return - Return Value Description
 public function getArtifactsOfImage(http:Request getImageRequest, string orgName, string imageName, string artifactVersion,
 int offset, int resultLimit) returns http:Response {
+    log:printDebug(io:sprintf("Listing artifacts for organization \'%s\' imageName : \'%s\', version: \'%s\', offset: %d, limit: %d",
+    orgName, imageName, artifactVersion, offset, resultLimit));
 
-    log:printDebug("Listing artifacts for organization \'" + orgName + "\' imageName : " + imageName + ", version: " + artifactVersion
-    + ", offset: " + offset + ", limit: " + resultLimit);
-
-    table<gen:ArtifactListResponse> | error artifactListResults;
+    table<gen:ArtifactList> | error artifactListResults;
 
     if (getImageRequest.hasHeader(constants:AUTHENTICATED_USER)) {
         string userId = getImageRequest.getHeader(constants:AUTHENTICATED_USER);
-        log:printDebug("List artifacts of image request with authenticated User : " + userId);
+        log:printDebug(io:sprintf("List artifacts of image request with authenticated User  : %s", userId));
         artifactListResults = db:getArtifactsOfUserImage(orgName, imageName, userId, artifactVersion, offset, resultLimit);
     } else {
         log:printDebug("List artifacts of image request without an authenticated user");
         artifactListResults = db:getArtifactsOfPublicImage(orgName, imageName, artifactVersion, offset, resultLimit);
     }
 
-    if (artifactListResults is table<gen:ArtifactListResponse>) {
-        log:printDebug("Number of results found for list image ======== : " + artifactListResults.count());
+    if (artifactListResults is table<gen:ArtifactList>) {
+        log:printDebug(io:sprintf("Number of results found for list image : %d", artifactListResults.count()));
         gen:ArtifactListResponse[] responseArray = [];
         int counter = 0;
+        int listLength = 0;
+        string artifactImageId = "";
         gen:ArtifactListArrayResponse response = {
-            count: 0,
+            count: counter,
             data: responseArray
         };
 
         if (artifactListResults.count() == 0) {
             log:printError("No image found with given image name and organization");
         } else {
             log:printError(io:sprintf("Found %d result(s) for artifact list", artifactListResults.count()));
-            string artifactImageId = "";
-            while (artifactListResults.hasNext()) {
-                gen:ArtifactListResponse result = <gen:ArtifactListResponse> artifactListResults.getNext();
-                responseArray[counter] = result;
+
+            foreach var item in artifactListResults {
+                gen:ArtifactList result = gen:ArtifactList.convert(item);
```

Comment:
gen:ArtifactList struct name seems to be referring to a single data item. The name should be changed accordingly.